CARDS-2474: Clinic dashboards are too slow on large datasets

sdumitriu opened the pull request:
To test:
- build this branch with `mvn clean install -Pdocker`
- in `compose-cluster` generate a docker compose file with `python3 generate_compose_yaml.py --mssql --cards_project cards4proms --oak_filesystem --dev_docker_image --composum --adminer --server_address=localhost:8080`
- in `compose-cluster/mssql` run `python3 ./generate_test_proms_sql.py -n 5000 proms_sample.sql` to generate a big sample file
- start with `docker-compose build && docker-compose up`
- at `http://localhost:1435/?mssql=mssql&username=sa&db=master&ns=path&import=` import the sample sql file from `compose-cluster/mssql`
- access `http://localhost:8080/Subjects.importClarity`  to import the visits
- check that the dashboard loads quickly and shows the correct results
- fill in and submit a couple of visits
- check that the clinic dashboard loads quickly, especially the past tabs
- stop and clean up with `docker-compose rm -vf && ./cleanup.sh`

sdumitriu committed:
CARDS-2474: Clinic dashboards are too slow on large datasets
Don't use any property index when running these queries

sdumitriu committed:
CARDS-2474: Clinic dashboards are too slow on large datasets
Don't count all matching rows in the clinic dashboard livetables

sdumitriu committed:
CARDS-2474: Clinic dashboards are too slow on large datasets
Mark that jcr:uuid properties are unique in the index

marta- reviewed (approved):
looks good, didn't test

sashaandjic commented:
A small issues here:
![image](https://github.com/data-team-uhn/cards/assets/23243244/79860590-faa3-45cb-bb1c-797b236134a0)

sashaandjic commented:
Speed:

https://github.com/data-team-uhn/cards/assets/23243244/4a9ce455-b1eb-4b18-a714-ad4bd21d0c16

sashaandjic commented:
Filtering Patients by e.g. name is very slow.

sashaandjic commented:
I completed and submitted surveys for one of the uploaded patients. The Visit info form does not seem to be updated correctly:
![image](https://github.com/data-team-uhn/cards/assets/23243244/8b685277-0205-4ea9-a4c5-9447cb4b619f)

sashaandjic commented:
Same issue as before - surveys submitted, but not completed
![image](https://github.com/data-team-uhn/cards/assets/23243244/032f79f6-10cf-440b-a4c8-6c317f7a5a24)

sashaandjic commented:
Speed:

https://github.com/data-team-uhn/cards/assets/23243244/6d6faddf-875b-4138-9df6-9ea1d8058db3

sdumitriu committed:
CARDS-2474: Clinic dashboards are too slow on large datasets
Instead of returning -1 when the total number of rows is not needed, do the same estimation that the pagination servlet does

sashaandjic reviewed (commented):
tested

Imported 4590 patients
![image](https://github.com/data-team-uhn/cards/assets/23243244/e8c46ae3-f237-4c0c-88d4-bc0e2b5272ab)

After the import performance:
- 20 seconds to load 4588 visits in `PMCC-ACHD/Today`
- 8 seconds loading it a second time
- submitted visits were marked as incomplete

Stop the instance and started it again:
- 9 seconds to load 4586 visits in `PMCC-ACHD/Today`
- 3 second loading it a second time
- submitted visits were marked as submitted
